messenger/handler_results.rst

```diff
@@ -0,0 +1,124 @@
+.. index::
+    single: Messenger; Getting results / Working with command & query buses
+
+Getting results
+---------------
+
+When a message is handled, the :class:`Symfony\\Component\\Messenger\\Middleware\\HandleMessageMiddleware`
+adds a :class:`Symfony\\Component\\Messenger\\Stamp\\HandledStamp` for each responding handler,
+which can be used to get their returned value:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        use Symfony\Component\Messenger\MessageBusInterface;
+        use Symfony\Component\Messenger\Stamp\HandledStamp;
+
+        $envelope = $messageBus->dispatch(SomeMessage());
+
+        // Get the last handled stamp and its returned value:
+        $handledStamp = $envelope->last(HandledStamp::class);
+        $handledStamp->getResult();
+
+This assumes there is only one configured handler or that only the last handler
+result is relevant. But you can also get the stamps for all handlers:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        use Symfony\Component\Messenger\MessageBusInterface;
+        use Symfony\Component\Messenger\Stamp\HandledStamp;
+
+        $envelope = $messageBus->dispatch(SomeMessage());
+        $handledStamps = $envelope->all(HandledStamp::class);
+
+A :class:`Symfony\\Component\\Messenger\\HandleTrait` also exists in order to ease
+leveraging a Messenger bus for synchronous needs.
+The :method:`Symfony\\Component\\Messenger\\HandleTrait::handle` method ensures
+there is exactly one handler registered and returns its result.
+
+Working with command & query buses
+----------------------------------
+
+The Messenger component can be used in CQRS architectures where command & query
+buses are central pieces of the application.
+See Martin Fowler's `article about CQRS`_ to learn more and :doc:`how to configure multiple buses </messenger/multiple_buses>`.
+
+As queries are usually synchronous and expect to be handled once, getting the
+result from the handler is a common need.
+
+You can directly leverage the ``HandleTrait`` trait in an action (or any caller):
```

Comment:
We don't seem to be using the "action" terminology in the rest of the documentation. Would a controller be a better example?

Reply:
Indeed? Changed to controller.